src/Saml2/Provider.php

```diff
@@ -255,14 +380,33 @@ protected function getAssertionConsumerServiceRoute(): string
         return Str::of($this->getConfig('sp_acs', 'auth/callback'))->ltrim('/');
     }
 
-    protected function getAssertionConsumerServiceBinding(): string
+    protected function getSingleLogoutServiceRoute(): string
+    {
+        return Str::of($this->getConfig('sp_sls'))->ltrim('/');
+    }
+
+    protected function getDefaultAssertionConsumerServiceBinding(): string
+    {
+        $default = $this->hasRouteBindingType(
+            $this->getAssertionConsumerServiceRoute(),
+            SamlConstants::BINDING_SAML2_HTTP_REDIRECT
+        ) ? SamlConstants::BINDING_SAML2_HTTP_REDIRECT : SamlConstants::BINDING_SAML2_HTTP_POST;
+
+        return $this->getConfig('sp_default_binding_method', $default);
+    }
+
+    protected function hasRouteBindingType(string $route, string $bindingType): bool
     {
-        return Arr::has(
-            Route::getRoutes()->getRoutesByMethod(),
-            'GET.'.$this->getAssertionConsumerServiceRoute()
-        ) ?
-            SamlConstants::BINDING_SAML2_HTTP_REDIRECT :
-            SamlConstants::BINDING_SAML2_HTTP_POST;
+        $methods = [
+            SamlConstants::BINDING_SAML2_HTTP_REDIRECT => 'GET',
+            SamlConstants::BINDING_SAML2_HTTP_POST     => 'POST',
+        ];
+
+        if (!array_key_exists($bindingType, $methods)) {
+            return false;
+        }
+
+        return Arr::has(Route::getRoutes()->getRoutesByMethod(), $methods[$bindingType].'.'.$route);
```

Comment:
This won't work when you have a route such as;
```
Route::match(['get', 'post'], '{provider}/callback', [SocialiteController::class, 'callback'])->name('callback');
```
(the relevant part being `{provider}`, ie. a "dynamic" url.

I know it was like before also, but maybe we can improve it here.

Something like this should work better;
```php
try {
  return Arr::has(
      app('router')->getRoutes()->match(app('request')->create($route, $methods[$bindingType]))->methods(),
      $methods[$bindingType]
  );
} catch(\Exception $e) {}

return false;
```

Only thing that doesn't work is if you have a "catch all" route.